store/src/main/java/com/dropbox/android/external/store4/Fetcher.kt

```diff
@@ -0,0 +1,49 @@
+package com.dropbox.android.external.store4
+
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.map
+
+typealias Fetcher<Key, Output> = (key: Key) -> Flow<FetcherResult<Output>>
+
+fun <Key : Any, Output : Any> fetcher(
+    doFetch: (Key) -> Flow<FetcherResult<Output>>
```

Comment:
do we not want to use invoke here? actually this might be OK in kotlin 1.4 as i think they did add SAM conversion for kotlin as well but i'm not 100% sure.

Reply:
Not sure what you mean. This function doesn't actually do anything it's just here to make the API more consistent

Reply:
oh nevermind.